pizza_cutter/coadding/_lanczos.py

```diff
@@ -116,29 +125,28 @@ def lanczos_resample_two(im1, im2, rows, cols, a=3):
 
         out_of_bounds = (
             x_f < 0 or
-            x_s > im1.shape[1]-1 or
+            x_s > nx-1 or
             y_f < 0 or
-            y_s > im1.shape[0]-1)
+            y_s > ny-1
+        )
 
         if out_of_bounds:
-            res1[i] = np.nan
-            res2[i] = np.nan
-            continue
-
-        # clip the kernel to the input image if needed
-        x_s = max(0, min(x_s, im1.shape[1]-1))
-        x_f = max(0, min(x_f, im1.shape[1]-1))
-        y_s = max(0, min(y_s, im1.shape[0]-1))
-        y_f = max(0, min(y_f, im1.shape[0]-1))
+            edge[i] = True
 
         # now sum over the cells in the kernel
         val1 = 0.0
         val2 = 0.0
         for y_pix in range(y_s, y_f+1):
+            if y_pix < 0 or y_pix > ny-1:
```

Comment:
Should we avoid this branch? Ditto for the one in the next loop.

Reply:
not sure if it matters, I can profile

Reply:
I tried instead clipping the y_s etc. and that was a lot slower

Reply:
weird!

Reply:
actually the new version seems to be considerably faster than the old one in master

Reply:
```
   Ordered by: internal time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
      202   51.522    0.255   52.131    0.258 _wcs_utils.py:55(__call__)
      183   43.500    0.238   48.077    0.263 interpolate.py:105(_grid_interp)
        9   31.991    3.555   32.120    3.569 _wcs_utils.py:50(__init__)
        6   29.661    4.944   29.661    4.944 {method 'create_image_hdu' of '_fitsio.FITS' objects}
      101   29.166    0.289   29.166    0.289 _lanczos.py:77(lanczos_resample_two)
       84   13.114    0.156   15.544    0.185 fitslib.py:120(read_header)
       48   10.440    0.218   10.440    0.218 {method 'read_image' of '_fitsio.FITS' objects}
       72    5.194    0.072    5.194    0.072 {method 'normal' of 'mtrand.RandomState' objects}
      664    4.425    0.007    4.440    0.007 wcsutil.py:1177(Apply2DPolynomial)
```

Reply:
lanczos is no longer at the top

Reply:
this is due to the max(min) etc. calls I think

Reply:
Ummmmmmm - I assumed under njit those were very fast if statements. Is numba somehow grabbing the gil with those?

Reply:
We should adjust the other lanczos interp as well.

Reply:
the only difference is to do the if statements and continue, instead of doing `x_s = max(0, min(x_s, im.shape[1]-1))` etc.

So maybe those max min etc. are just slower than a simple if

Reply:
I am wondering if those statements rewritten as if statements but outside the inner two loops is the best thing. shrug

Reply:
I put it inside so that the interpolation still gets a chance to do something with incomplete information